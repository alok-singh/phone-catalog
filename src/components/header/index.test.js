import React from "react";
import Header from "./index";
import { BrowserRouter } from "react-router-dom";
import { render, cleanup } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

const renderApp = (props) => {
  return render(
    <BrowserRouter>
      <Header {...props} />
    </BrowserRouter>
  );
};

beforeEach(() => {});

afterEach(() => {
  cleanup();
});

test("Header renders correctly", () => {
  const props = {
    list: [{ link: "link", title: "title" }],
  };

  const { container } = renderApp(props);
  expect(container.firstChild).toMatchSnapshot();
});
