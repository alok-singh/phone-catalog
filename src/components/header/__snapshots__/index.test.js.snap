// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  class="header-wrapper"
>
  <a
    class="header-item"
    href="/link"
  >
    <div
      class="header-text"
    >
      title
    </div>
  </a>
</div>
`;
